src/vidpf.rs

```diff
@@ -0,0 +1,735 @@
+    }
+
+    /// initial_proof returns the initial value of a Proof for the root level.
+    pub fn initial_proof(&self) -> Proof {
+        Proof::new(self.proof_size)
+    }
+
+    /// initial_seed returns the initial value of a Seed for the root level.
+    pub fn initial_seed(&self) -> Seed {
+        Seed::new(self.seed_size)
+    }
+
+    fn prg(&self, seed: &Seed) -> Sequence {
+        let dst = "100".as_bytes();
+        let mut prg = self.prng_ctor.new_prng(seed, dst, &self.bind_info);
+
+        let mut sl = self.initial_seed();
+        let mut sr = self.initial_seed();
+        let mut tl = ControlBit::from(0);
+        let mut tr = ControlBit::from(0);
+        sl.fill(&mut prg);
+        sr.fill(&mut prg);
+        tl.fill(&mut prg);
+        tr.fill(&mut prg);
+
+        Sequence(sl, tl, sr, tr)
+    }
+
+    fn convert(&self, seed: &Seed) -> (Seed, R) {
+        let dst = "101".as_bytes();
+        let mut prg = self.prng_ctor.new_prng(seed, dst, &self.bind_info);
+
+        let mut out_seed = self.initial_seed();
+        let mut value = R::new();
+        out_seed.fill(&mut prg);
+        value.fill(&mut prg);
+
+        (out_seed, value)
+    }
+
+    fn hash_one(&self, alpha: &[u8], level: usize, seed: &Seed) -> Proof {
+        let dst = "vidpf cs proof".as_bytes();
+        let mut binder = Vec::new();
+        binder.extend_from_slice(alpha);
+        binder.extend(level.to_le_bytes());
+        let mut prg = self.prng_ctor.new_prng(seed, dst, &binder);
+
+        let mut proof = self.initial_proof();
+        proof.fill(&mut prg);
+
+        proof
+    }
+
+    fn hash_two(&self, proof: &Proof) -> Proof {
+        let dst = "vidpf proof adjustment".as_bytes();
+        let seed = self.initial_seed();
+        let binder = &proof.0;
+        let mut prg = self.prng_ctor.new_prng(&seed, dst, binder);
+
+        let mut out_proof = self.initial_proof();
+        out_proof.fill(&mut prg);
+
+        out_proof
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+/// ServerID used to identify two aggregation servers.
+pub enum ServerID {
+    /// S0 is the first server.
+    S0 = 0,
+    /// S1 is the second server.
+    S1 = 1,
+}
+
+#[derive(Debug)]
+/// Share
+pub struct Share<R>
+where
+    R: Range,
+    for<'a> &'a R: RangeOps<R>,
+{
+    /// y is the sharing of the output.
+    pub y: R,
+    /// pi is a proof used to verify the share.
+    pub pi: Proof,
+}
+
+/// RangeOps
+pub trait RangeOps<R>: Sized + Add<Output = R> + Sub<Output = R> {}
+
+/// Range
+pub trait Range: Fill + PartialEq + ConditionallyNegatable + ConditionallySelectable
+where
+    for<'a> &'a Self: RangeOps<Self>,
+{
+    /// new
+    fn new() -> Self;
+    /// is_zero
+    fn is_zero(self) -> bool;
+}
+
+/// Fill populates a struct from a PRNG source.
+pub trait Fill {
+    /// fill reads as many bytes as needed from a PRNG source to fill the [Self] struct.
+    fn fill(&mut self, r: &mut Box<dyn RngCore>);
+}
+
+/// GetPRG
+pub trait GetPRNG {
```

Comment:
It looks like the purpose of this trait is to translate an XOF seed into an `RngCore`. However, `SeedStream` implements `RngCore`, so you should be able just use the seed stream instead of wrapping it.

For example, `hash_two()` can be rewritten as follows:
```rust
    use rand::Rng;

    fn hash_two(&self, proof: &Proof) -> Proof {
        let dst = "vidpf proof adjustment".as_bytes();
        let seed = self.initial_seed();
        let binder = &proof.0;
        let mut rng = XofTurboShake128::seed_stream(
            &crate::vdaf::xof::Seed(seed.0.try_into().unwrap()),
            dst,
            binder,
        );
        let mut out_proof = self.initial_proof();
        rng.fill(&mut out_proof.0[..]);

        out_proof
    }
```

Reply:
adopted, it wasn't clear whether all the Xof have the same security properties. Consider to add a marker to distinguish.